core/js/src/typespecs/prompt.ts

```diff
@@ -0,0 +1,122 @@
+import { z } from "zod";
+
+export const roleSchema = z.enum([
+  "system",
+  "user",
+  "assistant",
+  "function",
+  "tool",
+  "model",
+]);
+export type Role = z.infer<typeof roleSchema>;
+
+export const functionCallSchema = z.object({
+  name: z.string(),
```

Comment:
Should we add docs to all of these fields? Or if they mirror some other documented schema elsewhere on the internet, we could link to those docs in the toplevel prompt schema.